docs_src/microservices/core/data/details/DataRetention.md

```diff
@@ -4,24 +4,118 @@ title: Core Data - Data Retention and Persistent Caps
+## Usage
 
-## Introduction
+### Count-based Retention 
+You can define the `maxCap` and `minCap` to purge events like the original design for count-based retention.
+```
+ "device": {
+     "name": "device_int_autoevents",
+     ...
+     "profileName": "profile_int_resources",
+     "autoEvents": [
+         { 
+             "interval": "1s", "onChange": false, "sourceName": "INT16_1", 
+             "retention": {
+                "maxCap": 2000, 
+                "minCap": 1000, 
+                "duration": "0s"
+             }
+          }
+       ]
+}
+```
+In this case, the Core Data service check whether the event count exceeds 2000, if count >= 2000, then purge events to meet the `minCap` 1000. If `minCap` is -1, the service removes all old data.
+
+### Time-based Retention Without MinCap
+You can define the `duration` to purge events for time-based retention.
+```
+"device": {
+    "name": "device_int_autoevents",
+    ...
+    "profileName": "profile_int_resources",
+    "autoEvents": [
+        { 
+            "interval": "1s", "onChange": false, "sourceName": "INT16_1", 
+            "retention": {
+                "maxCap": -1, 
+                "minCap": -1, 
+                "duration": "24h"
+            }
+        }
+    ]
+}
+```
+In this case, the `minCap` is -1 which means we don't keep old data, then the Core Data purge events that the age(current time - event origin) is greater than "24h".
+
+### Time-based Retention With MinCap
+You can define the `duration` and `minCap` for time-based retention and also keep minimum expired events in DB. Note that the `minCap` is used to keep old data when purging events, even if the database contains new data, the process still keep old data. This is useful when the device unexpectedly shutdown, and we still want to keep some old data for tracing.
+```
+"device": {
+    "name": "device_int_autoevents",
+    ...
+    "profileName": "profile_int_resources",
+    "autoEvents": [
+        { 
+            "interval": "1s", "onChange": false, "sourceName": "INT16_1", 
+            "retention": {
+                "maxCap": -1, 
+                "minCap": 1000, 
+                "duration": "24h"
+            }
+        }
+    ]
+}
+```
+In this case, the Core Data service purges events that the age(current time - event origin) is greater than "24h" and leaves 1000 old events in DB. 
 
-For detailed information on the data retention see [Use Case for capping readings in Core Data](../../../../design/ucr/Core-Data-Retention.md).
+For More details, the Core Data service will find the latest 1000th event, if the 1000th event doesn't exist, the service skip purging, otherwise,  the service removes the expired data that is older than the 1000th event.
```

Comment:
```suggestion
The Core Data service identifies the most recent 1000th event. If the 1000th event does not exist, the service skips purging. Otherwise, it removes expired data older than the 1000th event.
```

Reply:
Updated.